lib/src/entrypoint.dart

```diff
@@ -779,6 +899,19 @@ class Entrypoint {
         windowsLineEndings ? serialized.replaceAll('\n', '\r\n') : serialized);
   }
 
+  /// Saves a list of concrete package versions to the `pubspec.yaml` file.
+  ///
+  /// Will use Windows line endings (`\r\n`) if a `pubspec.lock` exists, and
+  /// uses that.
+  void _savePubspec(YamlEditor editor) {
+    final windowsLineEndings = fileExists(pubspecPath) &&
+        detectWindowsLineEndings(readTextFile(pubspecPath));
+
+    final serialized = editor.toString();
+    writeTextFile(pubspecPath,
+        windowsLineEndings ? serialized.replaceAll('\n', '\r\n') : serialized);
```

Comment:
Since you're mutating existing file.. and it already contains `\r\n` in some places, you might be creating `\r\r\n`?? :D